### DISCUSSION TITLE
Categories for Cart Rules

### DISCUSSION THREAD

**Discussion by fall-en** [PrestaShop/PrestaShop#41072] [Ideas]
#### Is your feature request related to a problem?

It is very annoying to see an infinite list of Cart Rules without any kind of categorization system. It will be easy for vendors to have Vouchers sorted by categories.

Example: if I have a module that generate 10% discount vouchers every time a customer place an order, I would like to group them into a specific category so I can scroll them faster and be able to modify them quickly without loosing time by searching them one by one. At the moment I have to scroll full list of Cart Rules looking for all the Vouchers generated by that specific module.

Another way to use categories can be to disable/enable a group of vouchers. Just disable the category to disable all the vouchers inside that specific category. Vendor can additionally be able to do bulk modifications on all the vouchers inside that category: change expiry date, change conditions, change discount value, make shipping free etc

#### Describe the solution you'd like

When manually creating a Cart Rule it should be possible to select a category for that Voucher like when creating a Product.
If vouchers are generated by a module, there should be a possibility to select in that specific module which category to assign for all the vouchers generated by that module.

Taking as example "Customer follow-up" module, it should be possible to select by a drop down menu the category in which each generated voucher should be inserted:

- CANCELLED CARTS > Select in which category should the generated voucher be inserted
- RE-ORDER > Select in which category should the generated voucher be inserted
- BEST CUSTOMERS > Select in which category should the generated voucher be inserted
- BAD CUSTOMERS> Select in which category should the generated voucher be inserted

If no category is selected just put the voucher in root category (as per Products).

**Comment by hibatallahAouadni** [comment]
Hello @fall-en 

Thank you for your suggestion. The Product Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

Thank you
